packages/hoppscotch-ui/src/index.ts

```diff
@@ -0,0 +1,32 @@
+import { Plugin } from "vue"
+import * as components from "./components/"
+
+const HoppUI = { ...components }
+
+// Hopp UI will be available globally when in a browser environment
+if (typeof window !== "undefined") {
+  ;(window as any)["HoppUI"] = HoppUI
+}
+
+import "./assets/scss/styles.scss"
+import "virtual:windi.css"
+
+export type HoppUIPluginOptions = {
+  t?: (key: string) => string
+  onModalOpen?: () => void
+  onModalClose?: () => void
+}
```

Comment:
You need to add proper JSDoc comments defining what each field does for public types.